list_map.py

```diff
@@ -0,0 +1,178 @@
+        return self.map[self.origin.y + coordinate.y][self.origin.x + coordinate.x] == PointState.occupied
+
+    def expand_top(self, amount_of_new_rows):
+        """
+        @brief Method used to expand the map on the top.
+        @details This adds new rows to the map on the top.
+        @param amount_of_new_rows The amount of rows you want to add to the top.
+        @warning This method moves the origin.
+        """
+
+        self.origin.y += amount_of_new_rows
+        newRow = [PointState.unoccupied for col in self.map[0]]
+        for i in range(amount_of_new_rows):
+            self.map.insert(0, list(newRow))
+
+    def expand_bottom(self, amount_of_new_rows):
+        """
+        @brief Method used to expand the map on the bottom.
+        @details This adds new rows to the map on the bottom.
+        @param amount_of_new_rows The amount of rows you want to add to the bottom.
+        """
+
+        newRow = [PointState.unoccupied for col in self.map[0]]
+        for i in range(amount_of_new_rows):
+            self.map.insert(len(self.map)-1, list(newRow))
+
+    def expand_left(self, amount_of_new_cols):
+        """
+        @brief Method used to expand the map on the left.
+        @details This adds new columns to the map on the left.
+        @param amount_of_new_cols The amount of columns you want to add to the left for each row.
+        @warning This method moves the origin.
+        """
+
+        self.origin.x += amount_of_new_cols
+        for row in self.map:
+            for col in range(amount_of_new_cols):
+                row.insert(0, PointState.unoccupied)
+
+    def expand_right(self, amount_of_new_cols):
+        """
+        @brief Method used to expand the map on the right.
+        @details This adds new columns to the map on the right.
+        @param amount_of_new_cols The amount of columns you want to add to the right for each row.
+        """
+
+        for row in self.map:
+            for col in range(amount_of_new_cols):
+                row.insert(len(row) - 1, PointState.unoccupied)
+
+    def update_top_left_point(self, new_point):
+        """
+        @brief Method used to update the top_left_point.
+        @param new_point The point you want to check if it should be the new top_left_point
+        @details Updates the top_left_point if the new_point is outside the outer border of the current map on the top left.
+        """
+
+        if new_point.x < self.top_left_point.x:
+            self.expand_left(self.top_left_point.x - new_point.x)
+            self.top_left_point = CartesianCoordinate(
+                new_point.x, self.top_left_point.y)
+        if new_point.y < self.top_left_point.y:
+            self.expand_top(self.top_left_point.y - new_point.y)
+            self.top_left_point = CartesianCoordinate(
+                self.top_left_point.x, new_point.y)
+
+    def update_bot_right_point(self, new_point):
+        """
+        @brief Method used to update the BotRightPoint.
```

Comment:
consistency on the internal variable name `bot_right_point` why aren't you using snake_case as everywhere else?